google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/models/Cluster.java

```diff
@@ -32,6 +30,60 @@
  * in the instance.
  */
 public class Cluster {
+  public enum State {
+    /** The state of the cluster could not be determined. */
+    NOT_KNOWN(com.google.bigtable.admin.v2.Cluster.State.STATE_NOT_KNOWN),
+    /** The cluster has been successfully created and is ready to serve requests. */
+    READY(com.google.bigtable.admin.v2.Cluster.State.READY),
+    /**
+     * The cluster is currently being created, and may be destroyed if the creation process
+     * encounters an error. A cluster may not be able to serve requests while being created.
+     */
+    CREATING(com.google.bigtable.admin.v2.Cluster.State.CREATING),
+    /**
+     * The cluster is currently being resized, and may revert to its previous node count if the
+     * process encounters an error. A cluster is still capable of serving requests while being
+     * resized, but may exhibit performance as if its number of allocated nodes is between the
+     * starting and requested states.
+     */
+    RESIZING(com.google.bigtable.admin.v2.Cluster.State.RESIZING),
+    /**
+     * The cluster has no backing nodes. The data (tables) still exist, but no operations can be
+     * performed on the cluster.
+     */
+    DISABLED(com.google.bigtable.admin.v2.Cluster.State.DISABLED);
+
+
+    private final com.google.bigtable.admin.v2.Cluster.State proto;
+
+    /**
+     * Wraps the protobuf. This method is considered an internal implementation detail and not meant
+     * to be used by applications.
+     */
+    @InternalApi
+    public static State fromProto(com.google.bigtable.admin.v2.Cluster.State proto) {
+      for (State state : values()) {
+        if (state.proto.equals(proto)) {
+          return state;
+        }
+      }
+      return NOT_KNOWN;
```

Comment:
This was intentional. I didn't think users would care about the distinction. Do you have something in mind?

Reply:
For the following two inputs, "fromProto" returns the same NOT_KNOWN.
1) com.google.bigtable.admin.v2.Cluster.State.STATE_NOT_KNOWN
2) com.google.bigtable.admin.v2.Cluster.State.[NEW_STATE]

It bothers me because STATE_NOT_KNOWN is a validate state where server doesn't know the current cluster state in some cases. As for [NEW_STATE], it is something more serious, we probably need to throw an exception because we don't know how to handle it.

Reply:
It's really not that serious. It can happen  if we add a state and the users don't update their clients. If this code threw an exception then we would break old versions of this client.